In [21]:
import pandas as pd
import numpy as np
from scipy import stats

In [16]:
data = pd.read_csv("/content/Costomer+OrderForm.csv")

In [ ]:
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [ ]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [17]:
data = pd.DataFrame(index=['Error Free', 'Defective'],
                    data={'Phillippines':[271,29], 'Indonesia':[267,33], "Malta":[269,31], "India":[280,20]})

In [18]:
data

Phillippines  Indonesia  Malta  India
Error Free           271        267    269    280
Defective             29         33     31     20

In [19]:
from scipy.stats import chi2_contingency

In [22]:
stat, p, dof, expected = stats.chi2_contingency(data)

In [24]:
stat

3.858960685820355

In [26]:
p     #p-Value

0.2771020991233135

##Result : since the p-Value (0.277) is greater than alpha (0.05), hence we failed to Reject ho.
# The percentage of defect is same in all the centers.
